### PR TITLE
Make items list empty on opening after the data reset in SelectBox (T720642)

### PR DESCRIPTION
**Pull request by KsyushaM**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 1]
```diff
@@ -444,6 +444,32 @@ QUnit.module("functionality", moduleSetup, () => {
...
+
+        keyboardMock($input)
+            .focus()
+            .type("one")
+            .change();
+
+        this.clock.tick(500);
+        selectBox.option("opened", false);
+        selectBox.option("dataSource", null);
+        $("#selectBox .dx-dropdowneditor-button").trigger("dxclick");
```
Comment (DokaRus) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]:
```suggestion
        $element.find(`.${DX_DROP_DOWN_BUTTON}`).trigger("dxclick");
```

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 2]
```diff
@@ -444,6 +444,32 @@ QUnit.module("functionality", moduleSetup, () => {
         assert.equal(listItems.length, 0, "items is not yet loaded");
     });
 
+    QUnit.test("Items list should be empty after dataSource reseting", (assert) => {
+        var data = ["one", "two"];
+
```
Comment (DokaRus) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]:
```suggestion
const $element = $("#selectBox");
```

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 3]
```diff
@@ -444,6 +444,32 @@ QUnit.module("functionality", moduleSetup, () => {
...
+        const selectBox = $("#selectBox").dxSelectBox({
```
Comment (DokaRus) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]:
```suggestion
        const selectBox = $element.dxSelectBox({
```

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 4]
```diff
@@ -444,6 +444,32 @@ QUnit.module("functionality", moduleSetup, () => {
...
+            dataSource: data,
+            searchEnabled: true
+        }).dxSelectBox("instance");
+
+        assert.deepEqual(selectBox._list.option("items"), data);
+
+        const $input = selectBox.$element().find(toSelector(TEXTEDITOR_INPUT_CLASS));
```
Comment (DokaRus) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]:
```suggestion
        const $input = $element.find(toSelector(TEXTEDITOR_INPUT_CLASS));
```

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 5]
```diff
@@ -444,6 +444,32 @@ QUnit.module("functionality", moduleSetup, () => {
...
+        keyboardMock($input)
+            .focus()
+            .type("one")
+            .change();
+
+        this.clock.tick(500);
+        selectBox.option("opened", false);
+        selectBox.option("dataSource", null);
+        $("#selectBox .dx-dropdowneditor-button").trigger("dxclick");
+        this.clock.tick(500);
```
Comment (DokaRus) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: Propably we can remove this `clock.tick`. Check it please

**Review by DokaRus — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Pull request merged**